api/src/main/java/io/opencensus/trace/SpanBuilder.java

```diff
@@ -260,28 +234,121 @@ public Span startSpan() {
+  public final NonThrowingCloseable startScopedSpan() {
+    return new ScopedSpanHandle(startSpan());
+  }
+
+  /**
+   * Returns the name of the {@code Span}.
+   *
+   * @return the name of the {@code Span}.
+   */
+  protected final String getName() {
+    return name;
   }
 
-  private SpanBuilder(
-      SpanFactory spanFactory,
-      @Nullable Span parentSpan,
-      @Nullable SpanContext parentSpanContext,
-      boolean remoteParent,
-      String name) {
+  /**
+   * Returns the parent {@code Span}, or {@code null} if this is a root {@code Span} OR {@code
+   * getRemoteParentSpanContext()} returns not {@code null}.
+   *
+   * <p>Used by the implementation of the {@link #startSpan()}.
+   *
+   * @return the parent {@code Span}, or {@code null} if this is a root {@code Span} OR {@code
+   *     getHasRemoteParent()} returns {@code true}.
+   */
+  @Nullable
+  protected final Span getParentSpan() {
+    return parentSpan;
+  }
+
+  /**
+   * Returns the remote parent {@code SpanContext}, or {@code null} if this is a root {@code Span}
+   * OR {@code getParentSpan()} returns not {@code null}.
+   *
+   * <p>If not {@code null} the newly created {@code Span} will have a remote parent (in a different
+   * process).
+   *
+   * <p>Used by the implementation of the {@link #startSpan()}.
+   *
+   * @return the remote parent {@code SpanContext}, or {@code null} if this is a root {@code Span}
+   *     OR {@code getHasRemoteParent()} returns {@code false}.
+   */
+  @Nullable
+  protected final SpanContext getRemoteParentSpanContext() {
+    return remoteParentSpanContext;
+  }
+
+  /**
+   * Returns the {@link Sampler} to be used, or {@code null} if default.
+   *
+   * <p>Used by the implementation of the {@link #startSpan()}.
+   *
+   * @return the {@code Sampler} to be used, or {@code null} if default.
+   */
+  @Nullable
+  protected Sampler getSampler() {
+    return sampler;
+  }
+
+  /**
+   * Returns the parent links to be set for the {@link Span}.
+   *
+   * <p>Used by the implementation of the {@link #startSpan()}.
+   *
+   * @return the parent links to be set for the {@code Span}.
+   */
+  protected final List<Span> getParentLinks() {
+    return parentLinks != null ? parentLinks : Collections.<Span>emptyList();
+  }
+
+  /**
+   * Returns the record events option, or {@code null} if default.
+   *
+   * <p>See {@link Span.Options#RECORD_EVENTS} for more details.
+   *
+   * <p>Used by the implementation of the {@link #startSpan()}.
+   *
+   * @return the record events option, or {@code null} if default.
+   */
+  @Nullable
+  protected final Boolean getRecordEvents() {
+    return recordEvents;
+  }
+
+  /**
+   * Constructs a new {@code SpanBuilder}.
+   *
+   * @param parentSpan the parent {@code Span}.
+   * @param remoteParentSpanContext the remote parent {@code SpanContext}, the parent is in a
+   *     different process.
+   * @param name the name of the {@code Span}.
+   */
```

Comment:
"throws NullPointerException if name is null" ?

It looks like its callers document this already.